src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -91,6 +91,26 @@ namespace
         default: return 0; // default
         }
     }
+
+    boost::optional<BitTorrent::TorrentContentLayout> jsonValueToContentLayout(const QJsonValue &jsonVal)
+    {
+        if (jsonVal.isNull())
+            return {};
+
+        if (!jsonVal.isString())
+            return {};
+
+        const QString str = jsonVal.toString();
+        return Utils::String::parse(str, BitTorrent::TorrentContentLayout::Default);
+    }
+
+    QJsonValue contentLayoutToJsonValue(const boost::optional<BitTorrent::TorrentContentLayout> contentLayout)
+    {
+        if (!contentLayout)
+            return {QJsonValue::Null};
+
+        return Utils::String::serialize(*contentLayout);
```

Comment:
seems you can combine them:
```c++
return contentLayout ? {Utils::String::serialize(*contentLayout)} : {};
```

Reply:
>seems you can combine them:

No. Unless you explicitly specify type name:
```c++
return contentLayout ? QJsonValue {Utils::String::serialize(*contentLayout)} : QJsonValue {};
```